server/container_stats_list.go

```diff
@@ -15,5 +15,32 @@ func (s *Server) ListContainerStats(ctx context.Context, req *pb.ListContainerSt
 		recordOperation(operation, time.Now())
 		recordError(operation, err)
 	}()
-	return nil, fmt.Errorf("not implemented")
+
+	// This is an inefficient method, since the container will be resolved twice,
+	// once by the container list code and once by the GetContainerStats call.
+	containers, err := s.ListContainers(ctx, &pb.ListContainersRequest{
+		&pb.ContainerFilter{
+			Id:            req.Filter.Id,
+			PodSandboxId:  req.Filter.PodSandboxId,
+			LabelSelector: req.Filter.LabelSelector,
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	var allStats []*pb.ContainerStats
+
+	for _, container := range containers.Containers {
+		stats, err := s.ContainerStats(ctx, &pb.ContainerStatsRequest{ContainerId: container.Id})
```

Comment:
Can we add a helper function to avoid the double look up?

Reply:
Will do - I'l refactor out the ListContainers filter logic and ContainerStats.

Reply:
I got completely derailed by a number of other totally unrelated tasks. I should be able to quickly refactor this by worst case tomorrow :)

Reply:
No worries. Thanks :)